api/main_endpoints/util/discord-connection.js

```diff
@@ -0,0 +1,39 @@
+const User = require('../models/User');
+const axios = require('axios');
+const {
+  discordApiKeys
+} = require('../../config/config.json');
+
+function loginWithDiscord(code, email) {
+  return axios
+    .post('https://discord.com/api/oauth2/token',
+      `client_id=${discordApiKeys.CLIENT_ID}` +
+      `&client_secret=${discordApiKeys.CLIENT_SECRET}` +
+      `&grant_type=authorization_code&code=${code}` +
+      `&redirect_uri=${discordApiKeys.REDIRECT_URI}`)
+    .then(response => {
+      axios.get('https://discord.com/api/users/@me', {
+        headers: {
+          Authorization: `Bearer ${response.data.access_token}`,
+        }
+      })
+        .then(loginResponse => {
+          User.findOne({ email })
+            .then((user) => {
+              user.discordID = loginResponse.data.id;
+              user.save();
+              return true;
+            });
+        })
+        .catch(() => {
+          return false;
+        });
+    })
+    .catch(() => {
+      return false;
```

Comment:
are you trying to return `true`/`false` back to wherever `loginWithDiscord` was called? If so, this won't behave as you expected.

You should return a `Promise` for this instead. `resolve(true)` where u originally returned it and `reject(false)` where it was originally a `return`

See [`api/google_api/util/SceGoogleApiHandler.js:52`](https://github.com/SCE-Development/Core-v4/blob/55ad626d991b3048a917b63abb68620a89911dad/api/google_api/util/SceGoogleApiHandler.js#L52) for an example.